tsconfig.json

```diff
@@ -4,9 +4,9 @@
     "moduleResolution": "node",
     "declaration": true,
     "jsx": "react",
+    "strict": true,
     "target": "es2018",
     "noEmit": true,
     "outDir": "./dist/"
-  },
-  "files": ["src/index.tsx"]
+  }
```

Comment:
VS Code (and potentially other IDEs) simply look for the nearest tsconfig up from a source file for TS compiler options, so in order to apply strict mode to the tests I've removed the files filter from the base `tsconfig.json` and added `tsconfig.main.json` which extends it that is specifically for building the library.